dvc/fs/gs.py

```diff
@@ -1,228 +1,43 @@
-import logging
-import os.path
+import base64
 import threading
-from datetime import timedelta
-from functools import wraps
 
 from funcy import cached_property, wrap_prop
 
-from dvc.exceptions import DvcException
 from dvc.path_info import CloudURLInfo
-from dvc.progress import Tqdm
 from dvc.scheme import Schemes
 
-from .base import BaseFileSystem
+from .fsspec_wrapper import FSSpecWrapper
 
-logger = logging.getLogger(__name__)
 
-
-def dynamic_chunk_size(func):
```

Comment:
Underlying fsspec upload is a bit complicated then a just a call (due to their own chunking) and they use a static threshold. I couldn't seem to trigger it with the defaults (5 MiB) when uploading 8 x 100 MiB files (I was thinking it might bottleneck, but it didn't). So we might be just safe on it, without actually worrying to much.